lib/improver/blending/weighted_blend.py

```diff
@@ -466,6 +595,15 @@ def process(self, cube, weights=None):
             msg = ('Coordinate to be collapsed not found in cube.')
             raise CoordinateNotFoundError(msg)
 
+        coord_dim = cube.coord_dims(self.coord)
+        if not coord_dim:
+            raise ValueError(
+                '{} has no associated dimension'.format(self.coord))
```

Comment:
This error message could maybe say something about that we are trying to blend across the coord that is printed in this message, eg. `"Blending coordinate {} has no associated dimension"`. (Also whilst looking at this, I realised the docstring for self.coord isn't very good, and should probably say that this is the coord we are collapsing/blending over. This isn't your fault, but feel free to fix it if you feel like it :) )

Reply:
OK!